src/box/txn.c

```diff
@@ -836,7 +837,7 @@ txn_journal_entry_new(struct txn *txn)
 
 	struct xrow_header **remote_row = req->rows;
 	struct xrow_header **local_row = req->rows + txn->n_applier_rows;
-	bool is_sync = false;
+	bool is_sync = txn->is_sync;
```

Comment:
It seems you shouldn't allow to make fully-local transactions synchronous:
```
tarantool> box.begin({is_sync=true}) box.space.loc:replace{1} box.commit()
Assertion failed: (lsn > prev_lsn), function vclock_follow, file vclock.c, line 46.
```
`loc` is a local space in this example.

Let's better throw an error when trying to commit a fully local transaction with `is_sync = true` .

Reply:
Done.